multiapps-controller-core/src/test/java/org/cloudfoundry/multiapps/controller/core/cf/TaggingRequestFilterTest.java

```diff
@@ -0,0 +1,169 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+
+import org.cloudfoundry.multiapps.controller.core.util.ApplicationConfiguration;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.reactive.ClientHttpRequest;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.reactive.function.BodyInserter;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ExchangeFunction;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+
+import reactor.core.publisher.Mono;
+
+public class TaggingRequestFilterTest {
+
+    private static final String TEST_VERSION_VALUE = "1.58.0";
+    private static final String TEST_ORG_VALUE = "faceorg";
+    private static final String TEST_SPACE_VALUE = "myspace";
+    private HttpHeaders httpHeaders;
+    private ClientRequest clientRequest;
+    @Mock
+    private ExchangeFunction nextFilter;
+    @Mock
+    private ApplicationConfiguration configuration;
+
+    @Before
+    public void setUp() throws Exception {
+        httpHeaders = new HttpHeaders();
+        clientRequest = initializeClientRequest();
+        MockitoAnnotations.openMocks(this)
+                          .close();
+    }
+
+    private ClientRequest initializeClientRequest() {
+        return new ClientRequest() {
+
+            @Override
+            public Map<String, Object> attributes() {
+                return null;
+            }
+
+            @Override
+            public BodyInserter<?, ? super ClientHttpRequest> body() {
+                return null;
+            }
+
+            @Override
+            public MultiValueMap<String, String> cookies() {
+                return null;
+            }
+
+            @Override
+            public HttpHeaders headers() {
+                return httpHeaders;
+            }
+
+            @Override
+            public String logPrefix() {
+                return null;
+            }
+
+            @Override
+            public HttpMethod method() {
+                return null;
+            }
+
+            @Override
+            public URI url() {
+                return null;
+            }
+
+            @Override
+            public Mono<Void> writeTo(ClientHttpRequest arg0, ExchangeStrategies arg1) {
+                return null;
+            }
+        };
+    }
+
+    @Test
+    public void testInjectGenericValue() throws IOException {
+        TaggingRequestFilter testedFilter = new TaggingRequestFilter("1.58.0");
+        testedFilter.filter(clientRequest, nextFilter);
+        assertNotNull(clientRequest.headers());
```

Comment:
You can assign clientRequest.headers() to local field called "actualHeaders" and use it afterwads.